.devcontainer/Dockerfile

```diff
@@ -9,4 +9,8 @@ RUN apt-get update \
     && apt-get update \
     && apt-get install -y dotnet-sdk-2.1 \
     && apt-get install -y dotnet-sdk-3.1 \
+    && wget https://packages.microsoft.com/config/ubuntu/22.04/packages-microsoft-prod.deb -O packages-microsoft-prod.deb \
```

Comment:
:exclamation:  this image is based on Ubuntu 20.04
and you are installing ubuntu 22.04 packages? "It works" is not an excuse! Please get this sorted out.

Furthermore, I'd suggest to put some version on the multiple packages-microsoft-prod.deb that are stored on disk, here.

Reply:
Happy to sort @jkowalleck , would the sort be changing Ubuntu to 22.04 and installing all versions from the 22.04 microsoft source?
as dotnet 7 is not available on the 20.04 microsoft source.

Reply:
@jkowalleck  Also to mention NET Core 3.1 or 2.0 are not supported on Ubuntu 22.04 Because the distro only supports OpenSSL 3, so not sure how you would lay this out in your solution?
1 devcontainer per Ubuntu version?

Reply:
I leave the decision to @coderpatros.
I do not know much about the specifics of .net.

Reply:
@jkowalleck  have created a seprated dockerfile that runs on Ubuntu 22.04 with dotnet 7 and the older config i have left on  its own docker file, and defaulted the devcontainer config to the Ubuntu 22.04 docker file.
Let me know if this works

Reply:
@coderpatros needs to decide. 
I just pointed out some oddities.

Reply:
I think bumping the base image to 22.04 makes sense. It's the current LTS version. .NET Core 3.1 and 2.0 are not supported anymore. So I'm happy to do a major version bump and drop support for them.